Sprint 1/issue 12#build the home screen and navigation bar

STR1ND3R opened the pull request:
Added the navbar and the home screen with its controller

ErikGmz committed:
Home screen and navigation bar progress.

Co-authored-by: Alan Hernández <BlazeDex@users.noreply.github.com>
Co-authored-by: Loloxo <STR1ND3R@users.noreply.github.com>111
Co-authored-by: Luis Gutierrez <DevLuis123@users.noreply.github.com>

STR1ND3R committed:
Navbar

Added Navbar to aplication layout

Co-authored-by: Luis Gutierrez <luisgutierrez26489@gmail.com>
Co-authored-by: Alan Hernández <ahernandez102@ucol.mx>
Co-authored-by: ErikGmz <erikgomez198755@outlook.com>

STR1ND3R committed:
Home Screen

Applied styles to home screen view

Co-authored-by: Luis Gutierrez <luisgutierrez26489@gmail.com>
Co-authored-by: Alan Hernández <ahernandez102@ucol.mx>
Co-authored-by: ErikGmz <erikgomez198755@outlook.com>

MauricioNC commented on app/controllers/home_controller.rb:
```diff
@@ -0,0 +1,5 @@
+class HomeController < ApplicationController
+  def index
+
```
Remove this empty line

MauricioNC commented on app/views/events/index.html.erb:
```diff
@@ -1,9 +1,9 @@
 <% @events.each_with_index do |event, index| %>    
   <h2><%= "Event no. #{format("%02d", (index + 1).to_s)} #{event.title}" %></h2>
-  <div><%= link_to 'Show event details', event_path(event) %></div>
+  <div class=""><%= link_to 'Show event details', event_path(event) %></div>
```
If you won't add any class, just remove the property

MauricioNC commented on app/views/home/index.html.erb:
```diff
@@ -1,13 +1,8 @@
+<div class="w-screen flex justify-center mt-24">
+<h1>
+    Events managment system
+</h1>
+</div>
```
Indent correctly this part of your code

STR1ND3R replied:
done

MauricioNC commented on app/views/layouts/application.html.erb:
```diff
@@ -12,6 +12,16 @@
+        <%= image_tag("logo.png", class:"my-0 h-4/5") %>
+      </div>
+      <div class="mx-4 py-5">
+        <%= button_to "GetStarted", "events#index" , method: :get, class:"bg-black rounded py-5 px-2 text-white font-bold"%>
```
Don't user button_to for get requests, instead use link_to.
Use button_to only for delete and post requests

STR1ND3R replied:
done

MauricioNC commented on app/views/layouts/application.html.erb:
```diff
@@ -12,6 +12,16 @@
 
   <body class="prose">
+
+    <nav class="bg-black h-28 w-screen flex flex-row ">
+      <div class="mx-4 py-5">
+        <%= image_tag("logo.png", class:"my-0 h-4/5") %>
+      </div>
+      <div class="mx-4 py-5">
+        <%= button_to "GetStarted", "events#index" , method: :get, class:"bg-black rounded py-5 px-2 text-white font-bold"%>
+
+      </div>
+    </nav>
```
You can export the nav in a separate file

STR1ND3R replied:
done

STR1ND3R committed:
Code review changes

Removed empty lines, unnecesary class properties, created a new file for navbar, changed button_to for link_to.

Co-authored-by: Luis Gutierrez <luisgutierrez26489@gmail.com>
Co-authored-by: Alan Hernández <ahernandez102@ucol.mx>
Co-authored-by: ErikGmz <erikgomez198755@outlook.com>

DevLuis123 committed:
Merge branch 'main' into sprint-1/issue-12#Build_the_home_screen_and_navigation_bar

MauricioNC commented on app/views/events/edit.html.erb:
```diff
@@ -1,3 +1,5 @@
 <%= render 'form', event: @event %>
-<%= link_to 'Back', event_path(@event), class: 'btn-3 ml-5 py-1 no-underline' %>
+
+<%= link_to 'Back', event_path(@event), class: 'btn-3 ml-5 py-1 no-underline' %>
```
Add a new empty line

STR1ND3R replied:
done

MauricioNC commented on app/views/home/index.html.erb:
```diff
@@ -1,10 +1,7 @@
   Events management system
 </h1>
+  
+</div>
```
Add a new empty line

STR1ND3R replied:
done

MauricioNC commented on app/views/layouts/_navbar.html.erb:
```diff
@@ -0,0 +1,9 @@
+    <%= link_to "GetStarted", "events#index", class:"bg-black rounded py-5 px-2 text-white font-bold"%>
+
+  </div>
+</nav>
```
Add a new empty line

STR1ND3R replied:
done

MauricioNC commented on app/views/layouts/application.html.erb:
```diff
@@ -12,6 +12,7 @@
   </head>
 
   <body class="prose">
+    <%= render 'layouts/navbar' %>
```
You can add the navbar in shared folder like:

/views/shared/_navbar.html.erb

And render like: 

<%= render 'shared/navbar' %>

STR1ND3R replied:
done

juperameza commented on app/views/home/index.html.erb:
```diff
@@ -1,10 +1,7 @@
   Events management system
 </h1>
+  
+</div>
```
New line

STR1ND3R replied:
done

STR1ND3R committed:
CR changes

Added empty lines and moved navbar file to shared folder

Co-authored-by: Luis Gutierrez <luisgutierrez26489@gmail.com>
Co-authored-by: Alan Hernández <ahernandez102@ucol.mx>
Co-authored-by: ErikGmz <erikgomez198755@outlook.com>

juperameza commented on test/controllers/home_controller_test.rb:
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class HomeControllerTest < ActionDispatch::IntegrationTest
+  # test "the truth" do
+  #   assert true
+  # end
```
If you dont need thisdelete it 
Don leave white comments
```suggestion

```

STR1ND3R replied:
done

STR1ND3R committed:
Deleted comments

Deleted unnecessary comments

Co-authored-by: Luis Gutierrez <luisgutierrez26489@gmail.com>
Co-authored-by: Alan Hernández <ahernandez102@ucol.mx>
Co-authored-by: ErikGmz <erikgomez198755@outlook.com>

juperameza reviewed (approved):
LGTM
